plugins/BEdita/Core/src/Model/Action/GetObjectAction.php

```diff
@@ -51,21 +53,44 @@ protected function initialize(array $data)
      */
     public function execute(array $data = [])
     {
-        $conditions = [
+        // Prepare primary key.
+        $key = array_map([$this->Table, 'aliasField'], (array)$this->Table->getPrimaryKey());
+        $primaryKey = (array)$data['primaryKey'];
+        if (count($key) !== count($primaryKey)) {
+            $primaryKey = $primaryKey ?: [null];
+            $primaryKey = array_map(function ($key) {
+                return var_export($key, true);
+            }, $primaryKey);
+
+            throw new InvalidPrimaryKeyException(sprintf(
+                'Record not found in table "%s" with primary key [%s]',
+                $this->Table->getTable(),
+                implode($primaryKey, ', ')
```

Comment:
Should this be `implode(', ', $primaryKey)` ?

Why this change?
Should we check for multi column primary keys on objects?

Reply:
Fixed. It would work anyway because [`implode()` accepts parameters in either order](http://php.net/implode), but it's definitely way way better how you suggested.

This snippet has been basically copy-pasted from CakePHP's `Table::get()`. I had to duplicate this portion of code because it is impossible to chain multiple finders with `Table::get()`, and as a consequence I had to perform some basic checks on primary key.

We don't need to support multi-column primary keys on objects, but we need to block attempts to pass multiple values for `primaryKey`:
```php
(new GetObjectAction(['table' => TableRegistry::get('Objects')]))
    ->execute(['primaryKey' => [1, 2, 3]]);
// This throws:
// new InvalidPrimaryKeyException(
//     'Record not found in table "objects" with primary key [1, 2, 3]'
// );
```

Reply:
Agreed.